### PR TITLE
Add player sensitivity to getPickResult in Entity.

### PR DESCRIPTION
**Pull request by darkevilmac**
This change is to match the corresponding method in Block.

See #3800

### CODE REVIEW

File: patches/minecraft/net/minecraft/entity/Entity.java.patch [hunk 1]
```diff
@@ -1,6 +1,14 @@
 --- ../src-base/minecraft/net/minecraft/entity/Entity.java
 +++ ../src-work/minecraft/net/minecraft/entity/Entity.java
-@@ -84,7 +84,7 @@
+@@ -14,7 +14,6 @@
+ import net.minecraft.block.Block;
+ import net.minecraft.block.BlockFence;
+ import net.minecraft.block.BlockFenceGate;
+-import net.minecraft.block.BlockLiquid;
```
Comment (CrafterKina) [inline patches/minecraft/net/minecraft/entity/Entity.java.patch]: Don't remove it.

**Review by CrafterKina — CHANGES_REQUESTED** [review]

File: patches/minecraft/net/minecraft/entity/Entity.java.patch (hunk 1 above)
Comment (darkevilmac) [reply patches/minecraft/net/minecraft/entity/Entity.java.patch]: Not sure how that happened, I'll adjust the patch.

File: patches/minecraft/net/minecraft/entity/Entity.java.patch [hunk 2]
```diff
@@ -154,12 +162,7 @@
 +        return true;
 +    }
 +
-+    /**
-+     * Called when a user uses the creative pick block button on this entity.
-+     *
-+     * @param target The full target the player is looking at
-+     * @return A ItemStack to add to the player's inventory, empty ItemStack if nothing should be added.
-+     */
++    @Deprecated // Use player sensitive version below.
```
Comment (bs2609) [inline patches/minecraft/net/minecraft/entity/Entity.java.patch]: Add a `@deprecated use {@link #getPickedResult(RayTraceResult, EntityPlayer)}` javadoc tag here.
If you think this method can be removed in future, add a `// TODO: remove` comment here as well.

File: patches/minecraft/net/minecraft/entity/Entity.java.patch [hunk 3]
```diff
@@ -211,6 +214,18 @@
...
 +    }
 +
++    /**
++     * Called when a user uses the creative pick block button on this entity.
++     *
++     * @param target The full target the player is looking at
++     * @param player The player that wants to pick
++     * @return A ItemStack to add to the player's inventory, empty ItemStack if nothing should be added.
++     */
++    public ItemStack getPickedResult(RayTraceResult target, EntityPlayer player)
```
Comment (bs2609) [inline patches/minecraft/net/minecraft/entity/Entity.java.patch]: Give this a `@Nonnull` annotation (use full `javax.annotation` name if needed)

**Review by bs2609 — COMMENTED** [review]
Looks ok, couple of small points:

File: patches/minecraft/net/minecraft/entity/Entity.java.patch [hunk 4]
```diff
@@ -1,6 +1,14 @@
 --- ../src-base/minecraft/net/minecraft/entity/Entity.java
 +++ ../src-work/minecraft/net/minecraft/entity/Entity.java
-@@ -84,7 +84,7 @@
+@@ -10,6 +10,7 @@
+ import java.util.Random;
+ import java.util.Set;
+ import java.util.UUID;
++import javax.annotation.Nonnull;
```
Comment (mezz) [inline patches/minecraft/net/minecraft/entity/Entity.java.patch]: Avoid adding this import by using `@javax.annotation.Nonnull` where you need it.

File: patches/minecraft/net/minecraft/entity/Entity.java.patch [hunk 5]
```diff
@@ -154,12 +162,8 @@
 +        return true;
 +    }
 +
-+    /**
-+     * Called when a user uses the creative pick block button on this entity.
-+     *
-+     * @param target The full target the player is looking at
-+     * @return A ItemStack to add to the player's inventory, empty ItemStack if nothing should be added.
-+     */
++    @Deprecated //Use {@link #getPickedResult(RayTraceResult, EntityPlayer)}
```
Comment (mezz) [inline patches/minecraft/net/minecraft/entity/Entity.java.patch]: Please use a javadoc for this  so it looks this way:
```java
/**
 * Called when a user uses the creative pick block button on this entity.
 *
 * @param target The full target the player is looking at
 * @return A ItemStack to add to the player's inventory, empty ItemStack if nothing should be added.
 * @deprecated use {@link #getPickedResult(RayTraceResult, EntityPlayer)}
 */
 //TODO: Remove in 1.12
 @Deprecated
 public ItemStack getPickedResult(RayTraceResult target)
```

**Review by mezz — CHANGES_REQUESTED** [review]

File: patches/minecraft/net/minecraft/entity/Entity.java.patch (hunk 5 above)
Comment (darkevilmac) [reply patches/minecraft/net/minecraft/entity/Entity.java.patch]: Should I keep "Use getPickedResult"?

Nevermind, just saw you have that in the new docs.

Comment (mezz) [reply patches/minecraft/net/minecraft/entity/Entity.java.patch]: Yes, that part is included in the javadoc

File: patches/minecraft/net/minecraft/entity/Entity.java.patch [hunk 6]
```diff
@@ -302,9 +318,9 @@
 +    public boolean canTrample(World world, Block block, BlockPos pos, float fallDistance)
 +    {
 +        return world.field_73012_v.nextFloat() < fallDistance - 0.5F
-+            && this instanceof EntityLivingBase
-+            && (this instanceof EntityPlayer || world.func_82736_K().func_82766_b("mobGriefing"))
-+            && this.field_70130_N * this.field_70130_N * this.field_70131_O > 0.512F;
++                && this instanceof EntityLivingBase
++                && (this instanceof EntityPlayer || world.func_82736_K().func_82766_b("mobGriefing"))
++                && this.field_70130_N * this.field_70130_N * this.field_70131_O > 0.512F;
```
Comment (mezz) [inline patches/minecraft/net/minecraft/entity/Entity.java.patch]: It looks like these lines were indented accidentally.

**Review by mezz — CHANGES_REQUESTED** [review]

**Comment by mezz** [comment]
Please give an example of how your mod will use this new feature.

File: patches/minecraft/net/minecraft/entity/Entity.java.patch (hunk 6 above)
Comment (darkevilmac) [reply patches/minecraft/net/minecraft/entity/Entity.java.patch]: Intellij auto indented when I pasted from a clean source. Fun, generating patches with a fix.

**Comment by darkevilmac** [comment]
> Please give an example of how your mod will use this new feature.

Well the main reason for this is making it better match the features of a method in Block from vanilla.
To quote from the issue I made prior to making a PR (#3800)

"It allows for developers to block off certain pick results depending on different aspects of the player as well."

I personally would use it to reference a class I have associated with each player to store data about what part of the entity they're looking at. I already have this data stored, so having to raycast again seems like a bit of a waste, and would greatly simplify my codebase.

**Review by mezz — APPROVED** [review]

**Review by CrafterKina — APPROVED** [review]

File: patches/minecraft/net/minecraft/entity/Entity.java.patch [hunk 7]
```diff
@@ -211,6 +214,19 @@
...
 +    }
 +
++    /**
++     * Called when a user uses the creative pick block button on this entity.
++     *
++     * @param target The full target the player is looking at
++     * @param player The player that picked this entity
++     * @return A ItemStack to add to the player's inventory, empty ItemStack if nothing should be added.
++     */
++    @javax.annotation.Nonnull
```
Comment (mezz) [inline patches/minecraft/net/minecraft/entity/Entity.java.patch:224]: This annotation can be removed, everything is Nonnull return value by default in vanilla minecraft code.

File: patches/minecraft/net/minecraft/entity/Entity.java.patch [hunk 8]
```diff
@@ -211,6 +214,19 @@
...
++    public ItemStack getPickedResult(RayTraceResult target, EntityPlayer player)
```
Comment (mezz) [inline patches/minecraft/net/minecraft/entity/Entity.java.patch:225]: Should player be `@Nullable`? I'm trying to think of a case but I'm not sure.

**Review by mezz — CHANGES_REQUESTED** [review]

**Comment by mezz** [comment]
@darkevilmac please respond to my short review and update to solve the merge conflict.

**Comment by mezz** [comment]
@darkevilmac you have not responded in a long time so I am closing this PR. 
If you want to update this feel free to reopen.

**Pull request closed without merging**